test/export/test_export.py

```diff
@@ -4826,6 +4826,51 @@ def forward(self, x, repeat):
         exported = export(model, inputs).module()
         self.assertEqual(model(*inputs), exported(*inputs))
 
+    def test_dynamic_shapes_wrapped_with_shape_guards(self):
+        class Neuron(torch.nn.Module):
+            def __init__(self, n_dims: int = 5, n_targets: int = 3):
+                super().__init__()
+                self.linear = torch.nn.Linear(n_dims, n_targets)
+
+            def forward(self, x, y):
+                return torch.sigmoid(self.linear(x + y))
+
+        args = (torch.randn(2, 5), torch.randn(2, 5))
+        batch = torch.export.Dim.DYNAMIC
+
+        n = Neuron()
+        compiled = export(
+            n, args, dynamic_shapes=({0: batch}, {0: batch})
+        )
+        expected = n(*args)
+        mod = compiled.module()
+        got = mod(*args)
+        self.assertTrue(torch.allclose(expected, got))
+
+        class Wrapped(Neuron):
+            def forward(self, *args):
+                return super().forward(*args)
+
+        w = Wrapped()
+
+        if is_retracebility_test(self._testMethodName):
```

Comment:
Why is this ok to fail?

Reply:
This is probably some existing issue with how dynamic shapes are specified with var args.